### DISCUSSION TITLE
Intermittent API Connection Errors in Kreait\Firebase\Messaging\CloudMessage::sendAll

### DISCUSSION THREAD

**Discussion by nomori7010** [kreait/firebase-php#943] [Q&A]
Hello everyone,

I’ve been encountering intermittent errors while using the Kreait\Firebase\Messaging\CloudMessage::sendAll method to send messages. Occasionally, I see the following errors in the report's Failures section:

* "Unable to connect to the API: cURL error 35"
* "Unable to connect to the API: cURL error 28"
* An HTML error message like:
```
<!DOCTYPE html><html lang=en>  <meta charset=utf-8>  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">  <title>Error 502 (Server Error)!!1</title>  <style>    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}  </style>  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>  <p><b>502.</b> <ins>That’s an error.</ins>  <p>The server encountered a temporary error and could not complete your request.<p>Please try again in 30 seconds.  <ins>That’s all we know.</ins>
```

Why are these errors occurring? Has anyone else experienced this issue or found a solution? Any advice or insights would be greatly appreciated!

**Comment by yodelx** [comment]
Hello,

I have encountered the same error as well. Although it's not listed in the error code documentation, it seems there might be times when the FCM server temporarily returns an HTTP 502 (Bad Gateway) due to server-side issues.

**Comment by jeromegamez** [comment]
Are you using the latest 7.x version of the SDK? Earlier versions use the deprecated/shut-down legacy API of FCM and shouldn't work at all 🤔

**Comment by nomori7010** [comment]
I'm using SDK version 7.13. This issue occurs only rarely, not with every request.

**Reply by jeromegamez** [reply]
This looks like something on Firebase's side; I suggest creating a support request with them directly.

The thing is: if it works most of the time, but sometimes not, it's most likely not because of the SDK, and I can't fix this in code.
